lib/src/Uncertainty/Algorithm/WeightedExperiments/GaussProductExperiment.cxx

```diff
@@ -210,16 +210,16 @@ void GaussProductExperiment::save(Advocate & adv) const
 {
   WeightedExperimentImplementation::save(adv);
   adv.saveAttribute("collection_", collection_);
-  adv.saveAttribute("marginalDegrees_", marginalDegrees_);
+  adv.saveAttribute("marginalSizes_", marginalSizes_);
 }
 
 /* Method load() reloads the object from the StorageManager */
 void GaussProductExperiment::load(Advocate & adv)
 {
   WeightedExperimentImplementation::load(adv);
   adv.loadAttribute("collection_", collection_);
-  adv.loadAttribute("marginalDegrees_", marginalDegrees_);
-  setDistributionAndMarginalDegrees(distribution_, marginalDegrees_);
+  adv.loadAttribute("marginalSizes_", marginalSizes_);
```

Comment:
you'll want to provide a backward-compatible load path here:
```
if (adv.hasAtribute("marginalSizes_"))
  adv.loadAttribute("marginalSizes_", marginalSizes_);
else
adv.loadAttribute("marginalDegrees_", marginalDegrees_);
```

Reply:
Fine. Fixed.